### DISCUSSION TITLE
Drift on iOS with and without RTAB-Map

### DISCUSSION THREAD

**Discussion by FreakTheMighty** [introlab/rtabmap#1184] [Q&A] [answered]
I've been playing with the iOS application. I'm curious, are there any benchmarks comparing the drift/scale error seen in ARkit alone vs. what can be achieved by performing graph optimization or bundle adjustment using RTAB-Map?

If there aren't quantitative benchmarks, is there an overall expectation that the accuracy exceeds ARKit on its own?

**Comment by matlabbe** [comment]
There is no quantitative benchmark. The VIO pose should be exactly the same. The only difference would be how ARKit can deal with loop closures versus RTAB-Map. We assume from RTAB-Map side that pose returned by ARKit is smooth and continuous, without any jumps.

**Comment by FreakTheMighty** [comment]
@matlabbe I'm not sure I entirely understand. RTAB-Map is provided an initial set of poses from ARKit. When RTAB performs a loop closure would those poses be refined causing them to diverge from ARkit's originally reported poses? Additionally, when post processing a map via graph optimization, searching for additional loop closures, or bundle adjustment, at that point do poses diverge from ARKit's initial estimates?

**Reply by matlabbe** [reply, marked as answer]
Yes, after loop closures or post-optimization, the poses would likely be different between ARKit and RTAB-Map.
